platforms/core-configuration/declarative-dsl-core/src/main/resources/org/gradle/internal/declarativedsl/parsing/grammar.bnf

```diff
@@ -43,27 +48,29 @@
 
 root ::= statements
 
-statement ::= (assignment | block | functionChain)
+statement ::= (assignment | block | expression)
 statements ::= (statement SEMI*)*
 
-assignment ::= directlyAssignableExpression (EQ|PLUSEQ) assignmentValue {pin=2}
 directlyAssignableExpression ::= simpleIdentifier navigationSuffix?
```

Comment:
💭 By the way, do you recall why we are allowing the `navigationSuffix` here? Is it for `rootProject.name = ...`?

Reply:
Yes: https://docs.google.com/document/d/11GBpWfLV-BwYjGEayUl5OCoLzRlcDkI7m_b5FoaFD9Q/edit?disco=AAABNUxWlaI